src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Event/GetAzureRmRecoveryServicesAsrEvent.cs

```diff
@@ -0,0 +1,207 @@
+﻿// ----------------------------------------------------------------------------------
+// 
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Management.RecoveryServices.SiteRecovery.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Retrieves Azure Site Recovery alert and nofification settings.
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Get,
+        "AzureRmRecoveryServicesAsrEvent",
+        DefaultParameterSetName = ASRParameterSets.ByParam)]
+    [Alias("Get-ASREvent")]
+    [OutputType(typeof(IEnumerable<ASREvent>))]
+    public class GetAzureRmRecoveryServicesAsrEvents : SiteRecoveryCmdletBase
+    {
+        /// <summary>
+        ///     Gets or sets Resource Id.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = ASRParameterSets.ByResourceId,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        ///     Gets or sets Fabric Id.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByFabricId, Mandatory = true)]
+        public string FabricId { get; set; }
+
+        /// <summary>
+        ///     Gets or sets Event name.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByName, Mandatory = true)]
+        public string Name { get; set; }
+
+        /// <summary>
+        ///     Gets or sets server name.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByParam)]
+        [Parameter(ParameterSetName = ASRParameterSets.ByFabricId)]
+        [ValidateNotNullOrEmpty]
+        public string AffectedObjectFriendlyName { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the fabric.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByParam)]
+        [Parameter(ParameterSetName = ASRParameterSets.ByFabricId)]
+        [ValidateNotNullOrEmpty]
+        public ASRFabric Fabric { get; set; }
```

Comment:
@viverm raising this again - if a `Fabric` has a `FabricId`, why does the user need to provide one? It seems like you're just using it to get the `Name` of the fabric, but you already receive the name from the mandatory `Fabric` parameter that a user would pass.

Reply:
We can make is mutually exclusive and break in two paramset. If the user has fabricId or fabricObject he can directly call it.